### DISCUSSION TITLE
Determine if attribute is double or float before read

### DISCUSSION THREAD

**Discussion by wajihullahbaig** [BlueBrain/HighFive#606] [Questions] [answered]
Hi everyone
I have been trying figure out how to determine if the attribute I am reading is a double or float.
The code I have is as follows
```c++
 if (attribute.getDataType().getClass() == HighFive::DataTypeClass::Float)
                    {
                        double value;
                        attribute.read(value);
                    }
```
If I were to read the "value" variable as either float or double, how would can determine the data type before doing attribute.read(...)?

**Comment by alkino** [comment, marked as answer]
I think that you can get the size of the underlying type with:
https://bluebrain.github.io/HighFive/class_high_five_1_1_data_type.html#a2658ec3c1c3c77dadeb0b1b053d3a7d3

This way you can know if it is float (4 bytes) or double (8 bytes).

```c++
if (attribute.getDataType().getClass() == HighFive::DataTypeClass::Float)
{
    if (attribute.getDataType().getSize() == 4) {
        float value;
        attribute.read(value);
    } else if (attribute.getDataType().getSize() == 8) {
        double value;
        attribute.read(value);
    } else {
        std::cerr << "The float type size is not valid." << std::endl;
    }
}
```

**Reply by wajihullahbaig** [reply]
Thank you for the hint. I actually did it the same way you have suggested. 
Cheers!
